Sources/Client/Client/Client.swift

```diff
@@ -53,38 +53,50 @@ public final class Client {
     var waitingRequests = [WaitingRequest]()
     
     /// A web socket client.
-    lazy var webSocket = WebSocket()
+    lazy var webSocket: WebSocket = {
+        let webSocket = WebSocket()
+        webSocket.onConnect = setupWebSocketOnConnect
+        webSocket.onEvent = setupWebSocketOnEvent
+        return webSocket
+    }()
     
     public var connection: Connection { webSocket.connection }
     
     /// A WebSocket connection callback. This should only be used when you only use the Low-Level Client.
-    public var onConnect: Client.OnConnect = { _ in } {
-        didSet { webSocket.onConnect = setupWebSocketOnConnect }
-    }
+    public var onConnect: Client.OnConnect = { _ in }
     
     /// Saved onConnect for a completion block in `connect()`.
     var savedOnConnect: Client.OnConnect?
     
-    /// A WebSocket events callback. This should only be used when you only use the Low-Level Client.
-    public var onEvent: Client.OnEvent = { _ in } {
-        didSet { webSocket.onEvent = setupWebSocketOnEvent }
+    lazy var eventHandlingQueue: DispatchQueue = {
+        self.webSocket.webSocket.callbackQueue
+    }()
+    
+    var onEventObservers = [String: OnEvent]()
+    lazy var onEvent: Client.OnEvent = { [unowned self] event in
+        self.eventHandlingQueue.async {
+            self.onEventObservers.values.forEach({ $0(event) })
+        }
     }
```

Comment:
This is no needed anymore here as a var. You can move it to `setupWebSocketOnEvent`.